### PR TITLE
Dragging an item in the file tree allows scrolling upwards and downwards

### PR DESCRIPTION
**Pull request by theBadT**
This is an initial proposal for solving https://github.com/theia-ide/theia/issues/2918

Since I'm by far no typescript nor a react expert, feedback would be highly appreciated
@simark

### CODE REVIEW

File: packages/filesystem/src/browser/file-tree/file-tree-widget.tsx [hunk 1]
```diff
@@ -118,6 +118,15 @@ export class FileTreeWidget extends TreeWidget {
                 this.model.expandNode(containing);
             }
         }, 500);
+
+        let offset: number = 0;
+        if (event.clientY <= 90) {
+            offset = -20;
+        }
+        if (event.clientY >= 900) {
```
Comment (marechal-p) [inline packages/filesystem/src/browser/file-tree/file-tree-widget.tsx]: Magic numbers, they seem to only work with 1080p-ish windows.

Or maybe I just miss what the numbers mean, but the changes have an effect on a fullscreen browser window, but not when I resize to a smaller window.

Comment (marechal-p) [reply packages/filesystem/src/browser/file-tree/file-tree-widget.tsx]: To replace the magic number, maybe just a matter of finding the scrollable area size and computing the values from here?

`const bounds = event.currentTarget.getBoundingClientRect()` ?

**Comment by theBadT** [comment]
yeah, you're right. I already noticed that. thanks for the hint

File: packages/filesystem/src/browser/file-tree/file-tree-widget.tsx (hunk 1 above)
Comment (theBadT) [reply packages/filesystem/src/browser/file-tree/file-tree-widget.tsx]: found a way, thanks

**Comment by theBadT** [comment]
@marechal-p

**Comment by vince-fugnitto** [comment]
@theBadT I noticed when attempting to move a folder through the Explorer with many nodes that the scrolling is not continuous (there are abrupt animations). It'd be a lot nicer if it was more continuous and would provide a much better UX for end users.

scroll

As a reference, here is vscode:

sv

**Comment by theBadT** [comment]
yeah, you're right @vince-fugnitto looks pretty bad 😢 I'll do my very best

**Comment by vince-fugnitto** [comment]
> yeah, you're right @vince-fugnitto looks pretty bad  I'll do my very best

Don't worry you already did great ! It's 100% better already than it was previously 😊

**Comment by akosyakov** [comment]
@marechal-p @vince-fugnitto Is it good? Can be merged?

**Comment by vince-fugnitto** [comment]
@akosyakov no, its not yet ready, the scrolling hasn't been resolved https://github.com/theia-ide/theia/pull/4726#issuecomment-479012288  :(
@theBadT do you still plan on working on the issue?

**Comment by vince-fugnitto** [comment]
@theBadT any plans to continue working on the PR?

**Comment by akosyakov** [comment]
@vince-fugnitto maybe you take over, it is really annoying usability glitch :(

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto maybe you take over, it is really annoying usability glitch :(

I haven't had the time or luck in providing a fix for the d&d scrolling without the use of magic numbers. Perhaps someone like @jbicker would have more luck once they have a little spare time.

**Comment by vince-fugnitto** [comment]
Thank you for your contribution!
The pull-request will be closed due to inactivity, and since we want to keep a manageable number of open prs.
Please feel free to re-open the pull-request if you still plan on providing changes.

**Pull request closed without merging**